gatewayconfig/bluetooth/characteristics/wifi_connect_characteristic.py

```diff
@@ -30,23 +59,26 @@ def __init__(self, service):
     def check_wifi_status(self):
         # Check the current wi-fi connection status
         logger.debug('Check WiFi Connect')
-        nm_state = str(nmcli_custom.device.show('wlan0')['GENERAL.STATE'].split(" ")[0])
 
-        # Convert the network manager device state into wifi status response
-        wifi_status = constants.WIFI_STATUSES.get(nm_state,
-                                                  constants.WIFI_ERROR)
+        if self.connecting:
+            wifi_status = constants.WIFI_CONNECTING
+        else:
+            nm_state = str(nmcli_custom.device.show('wlan0')['GENERAL.STATE'].split(" ")[0])
```

Comment:
`wlan0` is not constant across devices. For example, I have `wlo1` on my laptop. We should evaluate the effort to make it parametric.

Reply:
Let me try to generalize this part.

Reply:
There are places like hm-diag also where we depend on eth0 and wlan0 being there. It might be worth a thought to make sure that they consistently there using udev rules, consistent-names etc.